client/src/components/graph/overlays/centroidLabels.js

```diff
@@ -1,14 +1,32 @@
+/* eslint-disable max-classes-per-file */
 /* eslint-disable jsx-a11y/mouse-events-have-key-events */
 import React, { PureComponent } from "react";
 import { connect } from "react-redux";
 
+import { categoryLabelDisplayStringLongLength } from "../../../globals";
+
 export default
 @connect(state => ({
   colorAccessor: state.colors.colorAccessor,
   dilatedValue: state.pointDilation.categoryField,
   labels: state.centroidLabels.labels
 }))
 class CentroidLabels extends PureComponent {
+  // Check to see if centroids have either just been displayed or removed from the overlay
+  componentDidUpdate = prevProps => {
+    const { labels, onDisplayChange } = this.props;
+    const prevSize = prevProps.labels.size;
+    const { size } = labels;
+
+    const displayChangeOff = prevSize > 0 && size === undefined;
+    const displayChangeOn = prevSize === undefined && size > 0;
+
+    if (displayChangeOn || displayChangeOff) {
+      // Notify overlay layer of display change
+      onDisplayChange("centroidLabels", displayChangeOn);
```

Comment:
This was surprising as a callback / event notation. I expected to see: `changeLabels` or something to that effect. Was there a different motivation?

Reply:
I wanted to try and keep the overlay layer as modular and child agnostic as possible.   I think in the future there's a chance that not every overlay component will have a redux action for on/off and there are also a few cases for centroids where toggling a category on doesn't necessarily mean any labels are displayed (toggling on a user annotation category with no values for example)